### PR TITLE
Make build automatic, reduce bundle size, add sourcemaps and test bundle

### PR DESCRIPTION
**Pull request by ariutta**
### Build

Changed `build` step so it works automatically:

```
git clone https://github.com/creatorrr/web-streams-polyfill.git
cd web-streams-polyfill
npm install
```

Before that change, I was getting errors because the `spec` directory was empty. I tried `cd ./spec && git pull origin master` and got a message saying it was up-to-date, but the directory was still empty. It would be nice to just add an entry `"whatwg-streams": "whatwg/streams#ba5cc06"` to `package.json`, but I was getting errors when I did that.
### Bundle Size

Changed the build process to use `uglifyify` (not just `uglify`) and `unassert` to reduce bundle size from 79K to 47K.
### Sourcemaps

Build process now generates sourcemaps.
### Tests

Originally, running `npm test` only tested the reference implementation code. Now it tests both the reference implementation and the bundled dist version.

### CODE REVIEW

**Comment by jonnyreeves** [comment]
Please can we get this landed @creatorrr; as it stands this module is unusable in the browser :(

**Comment by creatorrr** [comment]
Hey guys, sorry for the late reply. This branch seems to have merge conflicts. @ariutta Can you resolve them and update the request? I'll then merge and cut out a release asap.

File: package.json [hunk 1]
```diff
@@ -4,11 +4,15 @@
...
-    "test": "(cd spec/reference-implementation ; npm install && npm run test)",
-    "bundle": "browserify index.es6.js -s default -o dist/polyfill.js -t [ babelify ]",
-    "minify": "uglifyjs dist/polyfill.js -o dist/polyfill.min.js -c",
-    "build": "npm run bundle && npm run minify"
+    "test": "(cd spec/reference-implementation ; npm install && npm run test && cp ../../run-web-platform-tests-on-bundle.js ./ && node --expose_gc run-web-platform-tests-on-bundle.js && rm ./run-web-platform-tests-on-bundle.js)",
+    "bundle": "npm-run-all bundle:*",
+    "bundle:no-min": "browserify index.es6.js -d -s default -t [ babelify ] | exorcist ./dist/polyfill.js.map > ./dist/polyfill.js",
+    "bundle:min": "browserify -g uglifyify index.es6.js -d -i ./spec/reference-implementation/lib/utils.js -s default -t [ babelify --plugins babel-plugin-unassert uglifyify ] | exorcist ./dist/polyfill.min.js.map > ./dist/polyfill.min.js",
+    "build": "npm run bundle",
+    "prebuild": "rm -rf spec && git clone https://github.com/whatwg/streams.git && mv ./streams ./spec && cd spec && git checkout ba5cc069302db2e325815af7d2cd1e743e12773f && cd ..",
```
Comment (jimmywarting) [inline package.json]: Ain't this some Unix only commands that won't work in Windows?

**Comment by ariutta** [comment]
Hi all,

I'm afraid I won't be able to do this right away, so don't wait for me if
you want to get a release out ASAP.

Thanks!

**Comment by ariutta** [comment]
Linux and OS/X. Any suggestions for making it Windows compatible? I don't
use Windows.

**Comment by creatorrr** [comment]
@all Forking the windows build compatibility to a separate issue.
@ariutta No worries, I'll get this done.

**Comment by jimmywarting** [comment]
Actually... Why dose it have to build each time? Can't it just simply be prebuilt when publish?

**Comment by ariutta** [comment]
Yes, definitely. I just did it that way because I didn't have access rights
to publish.

**Comment by creatorrr** [comment]
Merged! I made some changes to revert to the submodule revision tracking and added your name to the contributors list, @ariutta

**Comment by creatorrr** [comment]
Also, the tests seem to be broken. :-/

**Comment by ariutta** [comment]
That's odd. I remember the tests passing for the version I submitted in the pull request. I did change `npm test`:

> Originally, running `npm test` only tested the reference implementation code. Now it tests both the reference implementation and the bundled dist version.

Is it possible your reversion could have affected that?

**Pull request merged**